filter.go

```diff
@@ -0,0 +1,39 @@
+package aiven
+
+import "errors"
+
+var (
+	_ Filter = (*BasicFilter)(nil)
+
+	// ErrUnableToFilter is the error message when the given elements cannot be filtered.
+	ErrUnableToFilter = errors.New("unable to filter given elements")
+)
+
+type (
+	// FilterableElements is a type that wraps the elements that can be filtered.
+	FilterableElements interface{}
+
+	// FiltrationResult is a type that wraps the result of filtering.
+	FiltrationResult interface{}
+
+	// FilterImplementation is a type that implements the Filter interface.
+	FilterImplementation struct{}
+
+	// Filter is a generic filter interface.
+	Filter interface {
+		// Filter filters the given elements.
+		Filter(fe FilterableElements) (FiltrationResult, error)
+	}
+
+	// BasicFilter is a basic filter implementation.
+	BasicFilter FilterImplementation
```

Comment:
Not sure why we need `BasicFilter` and `FilterImplementation` can you explain? 

It looks to me that we can simply implement interface for `K8sServiceTypesFilter` and `TfServiceTypesFilter` and assert it using `_ Filter = K8sServiceTypesFilter{}`

But perhaps I'm missing something.

Reply:
👍

Reply:
In the current implementation we are not using `BasicFilter` or `PassthroughFilter`, can you give a code snippet that puts it to work?

Reply:
this is no longer relevant